nav2_waypoint_follower/include/nav2_waypoint_follower/waypoint_follower.hpp

```diff
@@ -115,16 +126,62 @@ class WaypointFollower : public nav2_util::LifecycleNode
    */
   void goalResponseCallback(const rclcpp_action::ClientGoalHandle<ClientT>::SharedPtr & goal);
 
-  // Our action server
+  /**
+   * @brief send robot through each of GPS
+   *        point , which are converted to map frame first then using a client to
+   *        `FollowWaypoints` action.
+   *
+   * @param waypoints, acquired from action client
+   */
+  void followGPSWaypoints();
+
+/**
+ * @brief given some gps_waypoints, converts them to map frame using robot_localization's service `fromLL`.
+ *        Constructs a vector of stamped poses in map frame and returns them.
+ *
+ * @param gps_waypoints
+ * @param parent_node
+ * @param fromll_client
+ * @return std::vector<geometry_msgs::msg::PoseStamped>
+ */
+  static std::vector<geometry_msgs::msg::PoseStamped> convertGPSWaypointstoPosesinMap(
```

Comment:
I don't think this needs to be static. I don't think it would strictly speaking hurt, but I'm not sure it helps either.

Reply:
yes it isn't a requirement for this function to be static. But I think it can still be useful if someone wants to use this  function to convert their GPS waypoints to map without performing Waypoint Following and without need of creating an instance of `WaypoinFollower`. For example if they only want convert waypoints for visualization, then static feature of this function is  simpler to use.